apps/studio/components/interfaces/HomeNew/GettingStarted/GettingStartedSection.tsx

```diff
@@ -0,0 +1,563 @@
+
+type GettingStartedState = 'empty' | 'code' | 'no-code' | 'hidden'
+
+export function GettingStartedSection({
+  value,
+  onChange,
+}: {
+  value: GettingStartedState
+  onChange: (v: GettingStartedState) => void
+}) {
+  const { data: project } = useSelectedProjectQuery()
+  const { ref } = useParams()
+  const aiSnap = useAiAssistantStateSnapshot()
+  const router = useRouter()
+
+  // Local state for framework selector preview
+  const [selectedFramework, setSelectedFramework] = useState<string>(FRAMEWORKS[0]?.key ?? 'nextjs')
+  const workflow: 'no-code' | 'code' | null = value === 'code' || value === 'no-code' ? value : null
+
+  const { data: tablesData } = useTablesQuery({
+    projectRef: project?.ref,
+    connectionString: project?.connectionString,
+    schema: 'public',
+  })
+
+  const tablesCount = Math.max(0, tablesData?.length ?? 0)
+  const { data: branchesData } = useBranchesQuery({
+    projectRef: project?.parent_project_ref ?? project?.ref,
+  })
+  const isDefaultProject = project?.parent_project_ref === undefined
+  const hasNonDefaultBranch =
+    (branchesData ?? []).some((b) => !b.is_default) || isDefaultProject === false
+
+  // Helpers
+  const openAiChat = useCallback(
+    (name: string, initialInput: string) => aiSnap.newChat({ name, open: true, initialInput }),
+    [aiSnap]
+  )
+
+  const openConnect = useCallback(() => {
+    router.push(
+      {
+        pathname: router.pathname,
+        query: {
+          ...router.query,
+          showConnect: true,
+          connectTab: 'frameworks',
+          framework: selectedFramework,
+        },
+      },
+      undefined,
+      { shallow: true }
+    )
+  }, [router, selectedFramework])
+
+  const connectActions: GettingStartedAction[] = useMemo(
+    () => [
+      {
+        label: 'Framework selector',
+        component: (
+          <FrameworkSelector
+            value={selectedFramework}
+            onChange={setSelectedFramework}
+            items={FRAMEWORKS}
+          />
+        ),
+      },
+      {
+        label: 'Connect',
+        variant: 'default',
+        onClick: openConnect,
+      },
+    ],
+    [openConnect, selectedFramework]
+  )
+
+  const codeSteps: GettingStartedStep[] = useMemo(
+    () => [
+      {
+        key: 'install-cli',
+        status: 'incomplete',
+        title: 'Install the Supabase CLI',
+        icon: <Code strokeWidth={1} className="text-foreground-muted" size={16} />,
+        description:
+          'To get started, install the Supabase CLI to manage your project locally, handle migrations, and seed data.',
+        actions: [
+          {
+            label: 'Install via npm',
+            component: (
+              <CodeBlock className="w-full text-xs p-3 !bg" language="bash">
+                npm install supabase --save-dev
+              </CodeBlock>
+            ),
+          },
+        ],
+      },
+      {
+        key: 'design-db',
+        status: tablesCount > 0 ? 'complete' : 'incomplete',
```

Comment:
This might be sort of confusing given that it's mixing local and hosted flows. Since this is a local CLI Flow, 'design' should be considered complete once the user has a local migration file written (or pushed to their local DB), but this won't be complete until they have a table in their hosted DB.

Reply:
Code first can be broken down further into local vs remote code first. For local, you run local env etc. For remote, you still use remote database but cli for pushing/pulling. Starting with remote flow first, then can maybe break down into local later?